### ISSUE TITLE
ActionCable /cable mount exposes unfiltered redis password on Routing Error in development mode

### ISSUE THREAD

**Issue by mulander** [rails/rails#32055]
### Steps to reproduce

1. In `routes.rb`
```ruby
mount ActionCable.server => "/cable"
```
2. run `rails server`
3. Access a non existing URL or raise `ActionController::RoutingError.new('Not Found')` in one of your controllers.
4. Inspect the routes listed on the error page

(Guidelines for creating a bug report are [available
here](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#creating-a-bug-report))

### Expected behavior
I expect the 'Path' entry to match the console output of `rails routes`

which is:

```
                                 /cable                                                         #<ActionCable::Server::Base:0x000019a8ede4a8e0 @mutex=#<Monitor:0x000019a8ede4a890 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x000019a8ede4a7f0>>, @pubsub=nil, @worker_pool=nil, @event_loop=nil, @remote_connections=nil>
```

or even less.

### Actual behavior
The output contains a lot of data that I am not sure from where it is pulled, which among many others includes a `@raw_client` containing the redis connection string, password etc.

```
#>, @pubsub=#, @logger=#, @formatter=#, @logdev=#, @mon_owner=nil, @mon_count=0, @mon_mutex=#>, @local_levels=#>, @listener=#[#]}, @sync=#, @adapter=#, @event_loop=#, @executor=#, @__condition__=#, @min_length=1, @max_length=10, @idletime=60, @max_queue=0, @fallback_policy=:abort, @auto_terminate=true, @pool=[#, @pool=#, @thread=#>], @ready=[#, @pool=#, @thread=#>], @queue=[], @scheduled_task_count=2, @completed_task_count=2, @largest_length=1, @ruby_pid=82819, @gc_interval=30, @next_gc_time=56566.280579534, @StopEvent=#, @__condition__=#, @set=false, @iteration=0>, @StoppedEvent=#, @__condition__=#, @set=false, @iteration=0>>, @nio=#, @map={#=>#}, @stopping=false, @todo=#, @spawn_mutex=#>, @subscribe_callbacks={}, @subscription_lock=#, @raw_client=#"redis://test:test@127.0.0.1:6379/1", :scheme=>"redis", :host=>"127.0.0.1", :port=>6379, :path=>nil, :timeout=>5.0, :password=>"test", :db=>1, :driver=>Redis::Connection::Ruby, :id=>nil, :tcp_keepalive=>0, :reconnect_attempts=>1, :inherit_socket=>false, :role=>:master, :connect_timeout=>5.0, :read_timeout=>5.0, :write_timeout=>5.0, :_parsed=>true}, @reconnect=false, @logger=nil, @connection=#>, @command_map={}, @pending_reads=0, @connector=#"redis://test:test@127.0.0.1:6379/1", :scheme=>"redis", :host=>"127.0.0.1", :port=>6379, :path=>nil, :timeout=>5.0, :password=>"test", :db=>1, :driver=>Redis::Connection::Ruby, :id=>nil, :tcp_keepalive=>0, :reconnect_attempts=>1, :inherit_socket=>false, :role=>:master, :connect_timeout=>5.0, :read_timeout=>5.0, :write_timeout=>5.0, :_parsed=>true}>, @pid=82819>, @when_connected=[], @thread=#>, @redis_connection_for_broadcasts=nil>, @worker_pool=#, @__condition__=#, @min_length=1, @max_length=4, @idletime=60, @max_queue=0, @fallback_policy=:abort, @auto_terminate=true, @pool=[#, @pool=#, @thread=#>], @ready=[#, @pool=#, @thread=#>], @queue=[], @scheduled_task_count=2, @completed_task_count=2, @largest_length=1, @ruby_pid=82819, @gc_interval=30, @next_gc_time=56566.244362796, @StopEvent=#, @__condition__=#, @set=false, @iteration=0>, @StoppedEvent=#, @__condition__=#, @set=false, @iteration=0>>>, @event_loop=#, @executor=#, @__condition__=#, @min_length=1, @max_length=10, @idletime=60, @max_queue=0, @fallback_policy=:abort, @auto_terminate=true, @pool=[#, @pool=#, @thread=#>], @ready=[#, @pool=#, @thread=#>], @queue=[], @scheduled_task_count=2, @completed_task_count=2, @largest_length=1, @ruby_pid=82819, @gc_interval=30, @next_gc_time=56566.280579534, @StopEvent=#, @__condition__=#, @set=false, @iteration=0>, @StoppedEvent=#, @__condition__=#, @set=false, @iteration=0>>, @nio=#, @map={#=>#}, @stopping=false, @todo=#, @spawn_mutex=#>, @remote_connections=nil, @heartbeat_timer=#, @__condition__=#, @dup_on_deref=nil, @freeze_on_deref=nil, @copy_on_deref=nil, @do_nothing_on_deref=true, @execution_interval=3.0, @timeout_interval=30.0, @run_now=nil, @executor=#, @exception_class=StandardError, @__lock__=#, @__condition__=#>, @running=, @observers=#, @__condition__=#, @observers={}>, @StopEvent=#, @__condition__=#, @set=false, @iteration=0>, @StoppedEvent=#, @__condition__=#, @set=false, @iteration=0>>, @connections=[#[1, 3], "rack.errors"=>#>, "rack.multithread"=>true, "rack.multiprocess"=>false, "rack.run_once"=>false, "SCRIPT_NAME"=>"/cable", "QUERY_STRING"=>"", "SERVER_PROTOCOL"=>"HTTP/1.1", "SERVER_SOFTWARE"=>"puma 3.11.2 Love Song", "GATEWAY_INTERFACE"=>"CGI/1.2", "REQUEST_METHOD"=>"GET", "REQUEST_PATH"=>"/cable", "REQUEST_URI"=>"/cable", "HTTP_VERSION"=>"HTTP/1.1", "HTTP_HOST"=>"localhost:3000", "HTTP_CONNECTION"=>"Upgrade", "HTTP_PRAGMA"=>"no-cache", "HTTP_CACHE_CONTROL"=>"no-cache", "HTTP_UPGRADE"=>"websocket", "HTTP_ORIGIN"=>"http://localhost:3000", "HTTP_SEC_WEBSOCKET_VERSION"=>"13", "HTTP_USER_AGENT"=>"Mozilla/5.0 (X11; OpenBSD amd64; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.167 Safari/537.36", "HTTP_DNT"=>"1", "HTTP_ACCEPT_ENCODING"=>"gzip, deflate, br", "HTTP_ACCEPT_LANGUAGE"=>"pl-PL,pl;q=0.9,en-US;q=0.8,en;q=0.7", "HTTP_COOKIE"=>"CSRF-Token-PFQXQ=V4zbXJFuCRhqxyYZfg4Y5fxb5RSfnJsY; JSESSIONID.bd18d2c8=q0i8kqi6drwp9vfrxx1pt4yd; JSESSIONID.33842597=a9m39atoma8iudb72thj3vb4; user=\"!R6Y4HRcr6nb0nK/PCK5zmQ==?gAJVBHVzZXJxAVUIbXVsYW5kZXJxAoZxAy4=\"; JSESSIONID.4800428e=node02y8u5alxy0uh1t4fdv5wtlhti501.node0; JSESSIONID.51818afb=node080av052l1msa1vgepkzmfigtv0.node0; JSESSIONID.03c0cbf7=node01ke3l2ytlkpet83gw0upq9ph8103.node0; JSESSIONID.0ef72c18=node0uu19rqa90el0cesmcg7n5lrq1204.node0; JSESSIONID.9651af49=node0ivwrkmuswmsum6ouqg2n2r0p203.node0; cookie_eu_consented=true; JSESSIONID.f91ff1b8=node07f75qj74qe1dgar915209gf01190.node0; jenkins-timestamper-offset=-3600000; JSESSIONID.022e5fe7=node01ss248hww3ds41jed70h6yba8s1747.node0; _ga=GA1.1.726703923.1511802748; JSESSIONID.753887f2=node01jzd8f3ewscvwpaey9jqnz8i91293.node0; JSESSIONID.1ca1d740=node017ejq44c9rq2tz60tedaosp0y1020.node0; JSESSIONID.05171eb3=node0kvkd5zezlihg1ff85a7cj2xac944.node0; rack.session=BAh7C0kiD3Nlc3Npb25faWQGOgZFVEkiRWJlZmFkODM4Y2QyY2FiZTZiMmJi%0AMGI0OWM2MzIzZjkzMzEzYmY5OTU4NGUyYmNhNDY0NTRlZjc3NWMxZWIwZTgG%0AOwBGSSIZd2FyZGVuLnVzZXIudXNlci5rZXkGOwBUWwdbBmkCdEVJIiIkMmEk%0AMTEkRWdIUk9vSjJ0RHhDMzNxN1RvVFJjLgY7AFRJIh13YXJkZW4udXNlci51%0Ac2VyLnNlc3Npb24GOwBUewZJIhRsYXN0X3JlcXVlc3RfYXQGOwBUbCsH6tUu%0AWkkiEF9jc3JmX3Rva2VuBjsAVEkiMUNOZzdUczlDRkszaUNyaW5oNmVzNjc4%0AR3RHbEVXM3VucmcxTmVTV05MQzQ9BjsAVEkiCWNzcmYGOwBUSSIxVUMwNFNF%0AQlNLNWE5OHR2cVUzaFNPWGR4a0QyMG4zZ0lNdWFKMm9JS3Fvbz0GOwBUSSIN%0AdHJhY2tpbmcGOwBUewdJIhRIVFRQX1VTRVJfQUdFTlQGOwBUSSItNjUwZDI0%0AMDk5OTc2Mjg1NjlmZmNiZTcxNzY4MDFiNWI5YmYyY2RhOQY7AFRJIhlIVFRQ%0AX0FDQ0VQVF9MQU5HVUFHRQY7AFRJIi0yN2IwYjEzM2U0Zjk3YTRiNDUxMjVl%0ANWQyNDIzYjMxYzFiNjI2MWE3BjsAVA%3D%3D%0A--e7b5028d03eb9cdab6fa4bac77abbeea03a2d9dd; JSESSIONID.6c61bfed=node09946ze8tlib01es289blgtfsj211.node0; JSESSIONID.b420c6df=node0ws12hgpnxm34tn1bwejypobq375.node0; JSESSIONID.3137684c=node01gwkvfwrr2w0m1aq6vqit1aovx1164.node0; CSRF-Token-KKKWG=sNEPGxoW4oGW3msMhDeLjZMNukHHC4XH; JSESSIONID.1299014c=node0l7n2j0xelqsbv1rm27zign24473.node0; JSESSIONID.0b134960=node0nfbuo6xbznrwbyrvoo1iur7s504.node0; screenResolution=1366x768; JSESSIONID.c981a945=node0cvr0b1a8l5jgeok4aewhmifd1835.node0; CSRF-Token-DO4ZB=4wZuRfrF3kGNtFyYQbYVwQPRj63g3qCZ; firstVisit=1518042165253; JSESSIONID.5360cba0=node010616cjkvk2diunn800maes7z1995.node0; _webui_session=WXc4UXpwaldjOTVxNlBJc0FqdjRNUWZCSnhtdFgwejFqcEdrbmdTVlEzTllNRnBXVWc1a1RvNm82TC9RQlN0SFdYMnRRQ3dwZ0FnK0VYcG1nb0hwdUpLNXllYS9EcTJPTnhpMHZqMHF2Qjl1ZU5Nb0Z5Q0RjMDBJMlhxem1oN2pjNEM4Vk1DdU1KRlVtVnNPMnMwbUpZSTljeG1LczlEOHNLVWc1aUtKV1pDamdYMyt5Y1U5NCs0Umw1ZW1OTkh3WktCLzFvbUNpUURvSlI4ajdSd1lYUT09LS02SGV5ekRqZUFYQ21EU1NlTWxGMC9BPT0%3D--8574ba2bd4ffd43ed78ec05960c020ea4477cce7", "HTTP_SEC_WEBSOCKET_KEY"=>"LVsxRXMCTgKicmjNPGpYOQ==", "HTTP_SEC_WEBSOCKET_EXTENSIONS"=>"permessage-deflate; client_max_window_bits", "HTTP_SEC_WEBSOCKET_PROTOCOL"=>"actioncable-v1-json, actioncable-unsupported", "SERVER_NAME"=>"localhost", "SERVER_PORT"=>"3000", "PATH_INFO"=>"/", "REMOTE_ADDR"=>"127.0.0.1", "puma.socket"=>#, "rack.hijack?"=>true, "rack.hijack"=>#, "rack.input"=>#, "rack.url_scheme"=>"http", "rack.after_reply"=>[], "puma.config"=>#false, :app=>#:user, :scope_defaults=>{}, :default_strategies=>{:user=>[:rememberable, :database_authenticatable]}, :intercept_401=>false, :failure_app=>#}, @app=#, @ignore_cascade_pass=true>>, @cache_control="max-age=0, private, must-revalidate", @no_cache_control="no-cache">>>, @default_options={:path=>"/", :domain=>nil, :expire_after=>nil, :secure=>false, :httponly=>true, :defer=>false, :renew=>false}, @key="_webui_session", @cookie_only=true>>, @last_check=1512574008>>, @executor=#>, @routes_app=#, @response_format=:default>, @exceptions_app=#>>, @taggers=[]>, @assets_regex=/\A\/{0,2}\/assets/>, @check_ip=true, @proxies=[#, #, #, #, #, #]>>>, @header_name="X-Runtime">>, @executor=#>, @file_handler=#>>, @index="index">>, @variation="X-Accel-Redirect", @mappings=[]>, @config=#, @generators=#{:orm=>:active_record, :test_framework=>:test_unit, :integration_tool=>:test_unit, :system_tests=>:test_unit, :stylesheet_engine=>:scss, :javascript_engine=>:coffee}, :active_record=>{:migration=>true, :timestamps=>true}, :test_unit=>{:fixture=>true, :fixture_replacement=>nil}}, @fallbacks={}, @templates=[], @colorize_logging=true, @api_only=false, @hidden_namespaces=["sass"]>, @middleware=#, @encoding=#, @allow_concurrency=nil, @consider_all_requests_local=true, @filter_parameters=[:password], @filter_redirect=[], @helpers_paths=["/home/mulander/github/webui/app/helpers", "/home/mulander/github/webui/vendor/bundle/ruby/2.4/gems/local_time-2.0.0/app/helpers", "/home/mulander/github/webui/vendor/bundle/ruby/2.4/gems/devise-4.4.0/app/helpers", "/home/mulander/github/webui/vendor/bundle/ruby/2.4/gems/font-awesome-rails-4.7.0.3/app/helpers"], @public_file_server={:enabled=>true, :index_name=>"index"}, @force_ssl=false, @ssl_options={:hsts=>{:subdomains=>true}}, @session_store=:cookie_store, @time_zone="UTC", @beginning_of_week=:monday, @log_level=:debug, @cache_store=:null_store, @railties_order=[:all], @relative_url_root=nil, @reload_classes_only_on_change=true, @file_watcher=ActiveSupport::EventedFileUpdateChecker, @exceptions_app=nil, @autoflush_log=true, @log_formatter=#, @eager_load=false, @secret_token=nil, @secret_key_base=nil, @api_only=false, @debug_exception_response_format=nil, @x=#, @enable_dependency_loading=false, @read_encrypted_secrets=false, @paths=#, @root={"app"=>#, @glob="{*,*/concerns}", @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/assets"=>#, @glob="*", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "app/controllers"=>#, @glob=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/channels"=>#, @glob="**/*_channel.rb", @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/helpers"=>#, @glob=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/models"=>#, @glob=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/mailers"=>#, @glob=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/views"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=true>, "lib/assets"=>#, @glob="*", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib/tasks"=>#, @glob="**/*.rake", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/environments"=>#, @glob="development.rb", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/initializers"=>#, @glob="**/*.rb", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/locales"=>#, @glob="*.{rb,yml}", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/routes.rb"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db/migrate"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db/seeds.rb"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "vendor"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=true>, "vendor/assets"=>#, @glob="*", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/database"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/secrets"=>#, @glob="secrets.yml{,.enc}", @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/environment"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib/templates"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "log"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public/javascripts"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public/stylesheets"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "tmp"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/cable"=>#, @glob=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>}>, @autoload_paths=[], @eager_load_paths=["/home/mulander/github/webui/app/assets", "/home/mulander/github/webui/app/channels", "/home/mulander/github/webui/app/controllers", "/home/mulander/github/webui/app/controllers/concerns", "/home/mulander/github/webui/app/helpers", "/home/mulander/github/webui/app/jobs", "/home/mulander/github/webui/app/mailers", "/home/mulander/github/webui/app/models", "/home/mulander/github/webui/app/models/concerns", "/home/mulander/github/webui/app/uploaders", "/home/mulander/github/webui/app/workers", "/home/mulander/github/webui/app/channels/api_v1_channel.rb", "/home/mulander/github/webui/app/channels/benchmark_channel.rb", "/home/mulander/github/webui/app/channels/group_channel.rb", "/home/mulander/github/webui/app/channels/presence_channel.rb"], @autoload_once_paths=[], @cache_classes=false, @session_options={:key=>"_webui_session", :cookie_only=>true}>, @env_config={}, @helpers=nil, @routes=#, @app_build_lock=#, @initialized=true, @reloaders=[#, @files=#, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #, #}>, @dirs={#=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"], #=>["rb", "yml"]}, @block=#, @updated=, @lcsp=#, @pid=82819, @boot_mutex=#, @dtw=[#]>, #, #], @eager_load=false, @updater=#, @watched=nil, @updated_at=nil, @last_watched=["/home/mulander/github/webui/config/routes.rb", "/home/mulander/github/webui/vendor/bundle/ruby/2.4/gems/commontator-5.1.0/config/routes.rb"], @last_update_at=2017-12-12 16:28:13 +0100>>, #, @files=#, #}>, @dirs={#=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"], #=>["rb"]}, @block=#, @updated=, @lcsp=#, @pid=82819, @boot_mutex=#, @dtw=[#, #, #, #, #, #, #, #, #, #, #]>], @routes_reloader=#, #], @eager_load=false, @updater=#, @watched=nil, @updated_at=nil, @last_watched=["/home/mulander/github/webui/config/routes.rb", "/home/mulander/github/webui/vendor/bundle/ruby/2.4/gems/commontator-5.1.0/config/routes.rb"], @last_update_at=2017-12-12 16:28:13 +0100>>, @app_env_config={"action_dispatch.parameter_filter"=>[:password], "action_dispatch.redirect_filter"=>[], "action_dispatch.secret_token"=>nil, "action_dispatch.secret_key_base"=>"eacade364a977a4367271297899625ed84d06e246997cfc406b2b6260c3fad8236d228112451816f83a43712067564f751a545109294074914b25d8b76be2114", "action_dispatch.show_exceptions"=>true, "action_dispatch.show_detailed_exceptions"=>true, "action_dispatch.logger"=>#, @formatter=#, @logdev=#, @mon_owner=nil, @mon_count=0, @mon_mutex=#>, @local_levels=#>, "action_dispatch.backtrace_cleaner"=>#, #, #
--
```

since a picture is worth a thousand words
![2018-02-19-154829_1334x433_scrot](https://user-images.githubusercontent.com/107247/36383430-6a1e7ff8-158c-11e8-98ff-8ede91b17560.png)
![2018-02-19-154816_1253x449_scrot](https://user-images.githubusercontent.com/107247/36383432-6a453954-158c-11e8-86a0-99536ba21987.png)

### System configuration
**Rails version**:
```
fishtank$ ./bin/rails version                                                                                  
Rails 5.1.4
```
**Ruby version**:
```
fishtank$ ruby --version                                                                                       
ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-openbsd]
```

**Comment by y-yagi** [comment]
If you want to change mount path of Action Cable, please use `config.action_cable.mount_path`.

Also, I couldn't reproduce your issue. Are you using gems for error pages like `better_errors`?

**Comment by mulander** [comment]
Hi @y-yagi, no I don't want to change my mount paths. I only mentioned it to highlight how ActionCable is used within the app.

> Also, I couldn't reproduce your issue. Are you using gems for error pages like better_errors?

Are you using the built in `rails server`? I didn't see this occur on a passenger deployment so it might be related to the environment.

Regarding my gems & plugins, I'm attaching my Gemfile & Gemfile.lock.
[Gemfile.lock.txt](https://github.com/rails/rails/files/1738465/Gemfile.lock.txt)
[Gemfile.txt](https://github.com/rails/rails/files/1738466/Gemfile.txt)

**Comment by y-yagi** [comment]
Originally, Action Cable's mount path is treated as internal. As a result, mount path is not showed in the result of `routes`. 
https://github.com/rails/rails/blob/5-1-stable/actioncable/lib/action_cable/engine.rb#L46 

Yi\ou have explicitly mounted `ActionCable.server`, so the result has come out in routes. 
This is not the intended usage.. Therefore, the instance variable is showed unexpectedly. So please remove `mount ActionCable.server`. Thanks.

**Comment by mulander** [comment]
Thank you @y-yagi I see now what you hinted in your first reply. I removed the explicit mount and used `config.action_cable.mount_path` which removes `/cable` from the routes output as you mentioned.

I don't know where I got the idea to mount in routes.rb - might have been a very old version of http://guides.rubyonrails.org/action_cable_overview.html or some other site (still a lot of external materials point to mounting in routes.rb).

The issue is indeed solved, leaving this reply so anyone else hitting an outdated tutorial gets a reassurance.
